### PR TITLE
Use UriComponentsBuilder instead of HttpServletRequest in generatePag…

### PR DESCRIPTION
**Pull request by cbornet**
…inationHttpHeaders

It can then be used in resources like this
```java
    @GetMapping("/users")
    public ResponseEntity<List<UserDTO>> getAllUsers(@RequestParam MultiValueMap<String, String> queryParams, UriComponentsBuilder uriBuilder, Pageable pageable) {
        final Page<UserDTO> page = userService.getAllManagedUsers(pageable);
        HttpHeaders headers = PaginationUtil.generatePaginationHttpHeaders(uriBuilder.queryParams(queryParams), page);
        return new ResponseEntity<>(page.getContent(), headers, HttpStatus.OK);
    }
```
This code works for both MVC and Webflux.

### CODE REVIEW

File: jhipster-framework/src/test/java/io/github/jhipster/web/util/PaginationUtilUnitTest.java [hunk 1]
```diff
@@ -40,171 +41,142 @@
...
+        String expectedData = "</api/_search/example?query=Test%3E%3Btest&page=0&size=0>; rel=\"last\"," +
+            "</api/_search/example?query=Test%3E%3Btest&page=0&size=0>; rel=\"first\"";
         assertEquals(expectedData, headerData);
         List<String> xTotalCountHeaders = headers.get("X-Total-Count");
-        assertTrue(xTotalCountHeaders.size() == 1);
-        assertTrue(Long.valueOf(xTotalCountHeaders.get(0)).equals(0L));
-    }
-
-    @Test
-    public void sortAndCriteriaTest() {
```
Comment (cbornet) [inline jhipster-framework/src/test/java/io/github/jhipster/web/util/PaginationUtilUnitTest.java]: I didn't see the added value of this test so I removed it.

**Comment by cbornet** [comment]
See https://github.com/jhipster/generator-jhipster/pull/9379

File: jhipster-framework/src/test/java/io/github/jhipster/web/util/PaginationUtilUnitTest.java (hunk 1 above)
Comment (jdubois) [reply jhipster-framework/src/test/java/io/github/jhipster/web/util/PaginationUtilUnitTest.java]: this is to test the bug we had previously, you were losing the sort and criteria when navigating to another page

Comment (cbornet) [reply jhipster-framework/src/test/java/io/github/jhipster/web/util/PaginationUtilUnitTest.java]: Yes but the fix uses a generic way to add the query params. So it's already tested in the previous tests

**Comment by pvliss** [comment]
@jdubois + @DanielFran Is it ok with you guys if I merge this? LGTM and is required to get the CI back to green status. Current approach with `HttpServletRequest` is not working with webflux anyways. Once this is merged we can relaunch the CI build at jhipster/generator-jhipster#9379 to see if everything is working properly again. WDYT?

**Comment by cbornet** [comment]
Yes, this should fix the CI so let's merge.
@jdubois can you push the SNAPSHOT ?

**Pull request merged**